### PR TITLE
Allow for multiline opt description in __usage. Fixes #7

### LINKED ISSUE #7: Option flag description spanning more than a single line
It would be nice to be able to have better control over the usage statement. In particular:
- [ ] Option flag descriptions spanning more than one line
-  ~~More control over general usage statement formatting and content~~
-  ~~Additional, non option text, such as usage examples etc.~~

~~It's possible that the last point is a non-issue, I need to examine the usage text parsing more carefully.~~

The first point will need some different usage statement parsing logic. It's possible that some sort of special line break character or string could be used, and then when the usage statement is parsed for options, consecutive lines containing the special character/string would be concatenated, but when the usage statement is printed a line break is inserted at that character. An easy candidate could be `\n` which when `echo`ed with `echo -e` would insert a line break. Thoughts?

~~Not sure what to do about the second point yet...~~

### PR DESCRIPTION
**Pull request by mstreuhofer**
This PR fixes #7 which at first looked like a non-issue because everything seemed to work. But really multiple lines of description for an option was not supported. In certain cases it would even exit hard when trying to create variables with invalid names.

Now while I was at it I also made sure that defaults can be set for options only if they actually accept an argument.

If an option accepts an argument a default can be set on any line of the option's description. Last default wins.

### CODE REVIEW

**Comment by mstreuhofer** [comment]
Alright now that's weird. I want to say that this failing test cannot have anything to do with what was changed in this PR. The output seems fine. Just the color is missing. Might be a change on the Travis side which makes this line kick in?

`if [ "${NO_COLOR}" = "true" ] || [[ "${TERM:-}" != "xterm"* ]] || [ -t 1 ]; then`

**Comment by zbeekman** [comment]
Yes, I saw test failures caused by this change for OpenCoarrays and tweeted @travisci (support) and they told me to send an email with a link to a broken build to support@travis-co.com, which I haven't done yet due to work and being on the road (using my phone). Opencoarrays was failing somewhere because `TERM` was undefined and `set -o nounset` somewhere was bailing due to this. This is why it's not in color and the acceptance tests are expecting color.

Setting `TERM=xterm` in the Travis-CI project settings panel made the error go away for OpenCoarrays and think it will fix this error too.

**Comment by mstreuhofer** [comment]
@zbeekman @kvz Travis-CI checks return without error now. feels a little like windows. do nothing different, just retry and it works.

**Comment by mstreuhofer** [comment]
oh and btw the next PR is already waiting (but depends on this one here): automatic argument validation using {arg} vs [arg].

**Comment by gdevenyi** [comment]
Just chiming in that I have also had random Travis issues on another project running again usually resolves it.

**Comment by zbeekman** [comment]
@mstreuhofer yes, I confirmed that the TERM environment variable magically reappeared (and is set to xterm...)

**Comment by zbeekman** [comment]
LGTM, I'll see if @kvz has a chance to look at it, but if we don't hear from him for a while I'll merge this.

Thanks!

**Comment by kvz** [comment]
Would it make sense to support whatever Travis is setting TERM to, instead of changing that?

**Comment by zbeekman** [comment]
> Would it make sense to support whatever Travis is setting TERM to, instead of changing that?

@kvz Instead of changing what?

AFAIK, Travis sets TERM to xterm. I didn't specifically over ride it for b3bp, it is still being set by Travis. There was a hiccup with Travis, where they suddenly stopped setting TERM at all for a few builds. This has since been fixed on the Travis-CI end.

For b3bp, unsetting TERM has the potential to cause test failures in two ways:

1. As @mstreuhofer noted, when TERM is unset, then color output is turned off. Our acceptance tests require color output. So this is not a bug, but a feature: b3bp behaved correctly when it saw `$TERM != xterm*`.
2. TERM gets dereferenced somewhere without guarding against being undefined. This was happening with the forked version of b3bp in OpenCoarrays, but I have not seen this issue with the current b3bp.

### TL;DR: We are supporting what Travis sets `TERM` to, which is `xterm` and the test failure was due to a Travis-CI glitch that has since been fixed.

**Comment by kvz** [comment]
Ah okay, seems I misunderstood -- sorry about that! All good on my end then

**Comment by mstreuhofer** [comment]
so that's a maybe on the PR? ;)

**Comment by kvz** [comment]
Sorry if I'm a little absent minded, I'm very busy with tus.io and uppy.io at the moment. I'd be however happy to follow @zbeekman's judgement on this if that's okay.

**Comment by zbeekman** [comment]
Sounds good. I'm at a training course this week so I haven't had a chance
to give this a final review but I'll do so later today or tomorrow
